source/deprecations/v1.x.html.md

```diff
@@ -996,16 +996,44 @@ totalCount: Ember.computed('values.[]', function() {
 Because of changes in the way Ember renders components in 1.13, setting properties within an overridden `didInsertElement` method
 will result in a deprecation warning.
 
-In many cases you can move your logic later in the component lifecycle by implementing the `didRender` hook, on of
+In many cases you can move your logic earlier in the component lifecycle by implementing the `didReceiveAttrs` hook, one of
 [the new hooks introduced in 1.13](http://emberjs.com/blog/2015/06/12/ember-1-13-0-released.html#toc_component-lifecycle-hooks).
 
 ```js
-  didRender() {
+  didReceiveAttrs() {
     this._super(...arguments);
     this.set('myValue', value);
   }
 ```
 
+It may even be possible to move the change to `init`, where the value can only be set once.
+
+```js
+  init() {
+    this._super(...arguments);
+    this.set('myValue', myValue);
+  }
+```
+
+Don't see a `set` in your `didInsertElement`? It may be in `didUpdate` or `didRender`. Due to a bug, these run during the same process.
+
+Still can't find it? One of your computed properties may be calling `set`. Or one of your child components may be setting one of its attributes and it is being propagated upward to this component.
+
+In rare cases, you may be measuring the DOM rendered and want to set an attribute based on that. In this case, it is ok to cause a second render:
+
+```js
+  didInsertElement() {
+    this._super(...arguments);
+    run.scheduleOnce('afterRender', this, this.measureAndSet);
```

Comment:
Since `didInsertElement` is only called once, I don't think `render.scheduleOnce` is needed (though it isn't bad, it just takes a small amount more work to dedupe calls than `run.schedule`).